src/org/labkey/test/tests/list/ListDateAndTimeTest.java

```diff
@@ -0,0 +1,1368 @@
+import org.labkey.test.pages.list.EditListDefinitionPage;
+import org.labkey.test.params.FieldDefinition;
+import org.labkey.test.util.DataRegionExportHelper;
+import org.labkey.test.util.DataRegionTable;
+import org.labkey.test.util.ExcelHelper;
+import org.labkey.test.util.ListHelper;
+
+import java.io.File;
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+
+@Category({Daily.class, Data.class, Hosting.class})
+public class ListDateAndTimeTest extends BaseWebDriverTest
+{
+
+    private static final String PROJECT_NAME = "List Date And Time Test";
+    private SimpleDateFormat _defaultDateFormat = new SimpleDateFormat("yyyy-MM-dd");
+    private SimpleDateFormat _defaultTimeFormat = new SimpleDateFormat("HH:mm:ss");
+    private SimpleDateFormat _defaultDateTimeFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm");
+
+    @Override
+    public List<String> getAssociatedModules()
+    {
+        return Arrays.asList("list");
+    }
+
+    @Override
+    protected String getProjectName()
+    {
+        return PROJECT_NAME;
+    }
+
+    @BeforeClass
+    public static void setupProject() throws IOException, CommandException
+    {
+        ListDateAndTimeTest init = (ListDateAndTimeTest)getCurrentTest();
+        init.doSetup();
+    }
+
+    private void doSetup() throws IOException, CommandException
+    {
+        resetSiteSettings();
+        _containerHelper.createProject(PROJECT_NAME, null);
+        goToProjectHome();
+        LookAndFeelSettingsPage settingsPage = LookAndFeelSettingsPage.beginAt(this);
+
+        // Use java's Date object and SimpleDateFormatter (not strings) to enter the data.
+        _defaultDateFormat = new SimpleDateFormat(settingsPage.getDefaultDateDisplay());
+        _defaultTimeFormat = new SimpleDateFormat(settingsPage.getDefaultTimeDisplay());
+        _defaultDateTimeFormat = new SimpleDateFormat(settingsPage.getDefaultDateTimeDisplay());
+    }
+
+    private void resetSiteSettings() throws IOException, CommandException
+    {
+        log("Reset site settings.");
+        new SimplePostCommand("admin", "resetProperties")
+                .execute(createDefaultConnection(), "/");
+    }
+
+    @Before
+    public void preTest()
+    {
+        goToProjectHome();
+    }
+
+    @Override
+    protected BrowserType bestBrowser()
+    {
+        return BrowserType.CHROME;
+    }
+
+    private void validateListDataInUI(DataRegionTable table, List<Map<String, String>> expectedData)
+    {
+        checker().verifyEquals("Number of rows in the UI list not equal to number of expected rows.",
+                expectedData.size(), table.getDataRowCount());
+
+        int rowIndex = 0;
+        for(Map<String, String> expectedRowMap : expectedData)
+        {
+            // Protect against expecting more rows than are present in the grid.
```

Comment:
Why protect against this? You've already verified that the table has the correct number of rows.

Reply:
So it does not get an index out of bounds error and stops the entire tests.